core/services/llo/observation/data_source.go

```diff
@@ -180,7 +182,24 @@ func (d *dataSource) startObservationLoop(loopStartedCh chan struct{}) {
 		}
 
 		loopStart := time.Now()
-		opts, streamValues, observationInterval := d.getObservableStreams()
+		opts, streamValues, observationInterval, err := d.getObservableStreams()
+		// We might run into an error, or we might run into nil opts. In either case we can't continue.
+		if err != nil || opts == nil {
+			if err == nil {
+				err = errors.New("ObservableStreams opts is nil")
+			}
```

Comment:
I think it is OK to rely just on the error check since that is a standard pattern:
```suggestion
		if err != nil {
```

Reply:
Hmm. You're probably right. If somebody was to call `Observe()` with `nil` opts, we would have panic-ed already. Maybe I'm being overly defensive here.